Say/ViewController.swift

```diff
@@ -128,5 +134,22 @@ class ViewController: NSViewController {
         catch {/* error handling here */}
     }
     
+    @IBAction func setAlarm(_ sender: NSControl) {
+        if alarmButton.state == NSOnState {
+            self.alarmTime = datePicker.dateValue
+            self.timer = Timer(fireAt: alarmTime, interval: 0, target: self, selector: #selector(doAlarm), userInfo: nil, repeats: false)
+            RunLoop.main.add(timer, forMode: RunLoopMode.commonModes)
+        } else if alarmButton.state == NSOffState {
+            self.timer.invalidate()
+        }
+    }
+    
+    func doAlarm() {
+        alarmButton.isEnabled = false
+        SayAPI(text: self.textForSpeech, voice: self.selectedVoice).play(false)
+        alarmButton.setNextState()
```

Comment:
Alarm button is being checked when the program is saying something, so at the time the speech is ended `setNextState()` function **set Alarm button to be unchecked**.